### PR TITLE
Card status widget

### LINKED ISSUE #213: CardDetails page: Add a widget that displays the card's status (eg "Complete" or "In Review")

### PR DESCRIPTION
**Pull request by kaleem99**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kaleem99** [comment]
Tilde-card-status-widget

File: frontend/src/components/regions/CardDetails/Presentation.js [hunk 1]
```diff
@@ -17,23 +33,47 @@ import ProjectDetails from "./ProjectDetails";
...
-    maxHeight: 200,
-  },
-  sectionPaper: {
-    padding: theme.spacing(1),
-    marginBottom: theme.spacing(1),
-  },
-}));
+    [BLOCKED]: { ...card, backgroundColor: grey[400] },
+    [READY]: { ...card, backgroundColor: blue[400] },
+    [IN_PROGRESS]: { ...card, backgroundColor: green[400] },
```
Comment (sheenarbw) [inline frontend/src/components/regions/CardDetails/Presentation.js]: Are you sure this is DRY? It looks very familiar. 

If a piece of code needs to be reused then copy-paste usually isn't the best bet.  Rather make a little widget we can use  in multiple places

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Hmm this works but it's not DRY

**Review by Sbonelo01 — CHANGES_REQUESTED** [review]
Looking good @kaleem99 . Can you please work on the padding of the chip? Align it nicely with the ones above it

**Comment by babalwa-mbolekwa** [comment]
hey @kaleem99 can you also please attach the screenshot of the latest updated work

**Comment by kaleem99** [comment]
Screenshot 2021-11-09 132638

File: frontend/src/stories/fixtures/repoProjectCard.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "id": 13095,
   "contentItem": 223,
   "contentItemUrl": "https://raw.githubusercontent.com/Umuzi-org/tech-department/master/content/projects/oop/person/_index.md",
-  "status": "RF",
+  "status": "IP",
```
Comment (sheenarbw) [inline frontend/src/stories/fixtures/repoProjectCard.json]: Can you try not to change fixtures unless you absolutely need to? Sometimes they are set up in a specific way for a reason

**Review by babalwa-mbolekwa — CHANGES_REQUESTED** [review]
hey Kaleem well done. Looks good on stoybook, on the localhost it says `Loading...` and doesn't show anything on the board. Could you please double check if the Presentation files are correctly written

**Review by babalwa-mbolekwa — APPROVED** [review]
Well done Kaleem! It looks good on storybook. I couldn't check on the localhost because it displays `Loading...`

**Review by Vuyisanani — APPROVED** [review]

**Comment by Sbonelo01** [comment]
If the widget is supposed to be positioned like that, then all is good

File: frontend/src/components/regions/CardDetails/Presentation.js [hunk 3]
```diff
@@ -73,6 +84,9 @@ const CardBasicDetails = ({ card }) => {
           <FlavourChips flavourNames={card.flavourNames} />
           <StoryPoints storyPoints={card.storyPoints} />
         </Grid>
+        <TableCell>
+          <CardStatusChip card={card} />
+        </TableCell>
```
Comment (sheenarbw) [inline frontend/src/components/regions/CardDetails/Presentation.js]: It's a little odd to use a tablecell here. Where is the rest of the table?  Dont you want to use Grid things rather?

File: frontend/src/components/regions/GroupCardSummary/CellContent/Presentation.js [hunk 4]
```diff
@@ -115,9 +87,7 @@ export default ({ card, handleClickOpenCardDetails }) => {
               </TableCell>
               <TableCell>
                 <Paper className={classes[card.status]} variant="outlined">
```
Comment (sheenarbw) [inline frontend/src/components/regions/GroupCardSummary/CellContent/Presentation.js]: Are you sure this Paper shopuld still be here? It looks like it is a repeat of what is inside frontend/src/components/widgets/CardStatusChip.js and the classname doesn't make sense.

There are a lot of people on the team, it's best if everyone does their bit to keep things tidy.

File: frontend/src/components/widgets/CardStatusChip.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    [REVIEW_FEEDBACK]: { ...card, backgroundColor: red[400] },
+    [IN_REVIEW]: { ...card, backgroundColor: orange[400] },
+    [COMPLETE]: { ...card, backgroundColor: yellow[400] },
+  };
+});
+
+export default ({ card }) => {
+  const classes = useStyles();
+  return (
+      <Paper className={classes[card.status]} variant="outlined">
```
Comment (sheenarbw) [inline frontend/src/components/widgets/CardStatusChip.js]: The alignment here is a little strange. Can you please try to keep things tidy?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
This works but it has a few weird things slipping in. Some stuff is just a bit messy. And also it would be good to learn about what html tables do. Materiual ui tables are conceptually very similar to standard html <tables>. You shouldn't just use table cells oputside a table.

**Comment by kaleem99** [comment]
card-status

**Comment by kaleem99** [comment]
card-status2

**Review by babalwa-mbolekwa — CHANGES_REQUESTED** [review]
Could you please also double check the imports here also for material UI. It displays this error on the frontend
errorscolor

**Comment by kaleem99** [comment]
card-status

**Comment by kaleem99** [comment]
card-status2

**Review by babalwa-mbolekwa — APPROVED** [review]

**Review by Sbonelo01 — APPROVED** [review]
Looks good

**Review by Vuyisanani — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]
Weeee

**Pull request merged**